addons/auth_totp/controllers/home.py

```diff
@@ -34,6 +34,8 @@ def web_totp(self, redirect=None, **kwargs):
                     scope="browser", key=key, uid=user.id)
                 if user_match:
                     request.session.finalize(request.env)
+                    request.update_env(user=request.session.uid)
+                    request.update_context(**request.session.context)
```

Comment:
For consistency with `http.Session.authenticate()`
```suggestion
                    request.env = odoo.api.Environment(request.env.cr, request.session.uid, request.session.context)
                    request.update_context(lang=get_lang(request.env).code)
```

Reply:
@Julien00859 13 lines below the same two lines appear (`request.update_env...` and `request.update_context...`) would you like me to change those two lines as part of this PR too?

Reply:
Also, why updating only `lang` and not all of `request.session.context`?

Reply:
`request.env = ` already set all the context. We only change the lang because we wanna be certain that it is still active/valid. I'm wondering if it is necessary, might not be as `finalize` call `env['res.users'].context_get()` which should already provide a lang that is guaranteed to be installed...

Meh... ok, let's keep your initial suggestion as the same code exist down there already.